### PR TITLE
ISPN-7517 NettyTransport.stop() should wait for the workers threads to stop

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-7517
https://issues.jboss.org/browse/ISPN-7556
https://issues.jboss.org/browse/ISPN-7563
https://issues.jboss.org/browse/ISPN-7564
https://issues.jboss.org/browse/ISPN-7566

### CODE REVIEW

**Comment by tristantarrant** [comment]
Needs rebase, and also there are several related failures/.

**Comment by danberindei** [comment]
@tristantarrant There were quite a few failures, but I think I fixed all of them.

**Comment by danberindei** [comment]
Oops, the HotRod client test suite also has some problems...

**Comment by danberindei** [comment]
I fixed some more failures related to ISPN-7517 and ISPN-7566.

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HitsAwareCacheManagersTest.java [hunk 1]
```diff
@@ -200,23 +201,27 @@ protected void destroy() {
...
 
-      private volatile int invocationCount;
+      private final AtomicInteger invocationCount = new AtomicInteger(0);
 
       @Override
       protected Object handleDefault(InvocationContext ctx, VisitableCommand command) throws Throwable {
          if (ctx.isOriginLocal()) {
-            log.infof("Increment hit count after being visited by %s", command);
-            invocationCount ++;
+            if (Thread.currentThread().getName().contains("NodeB")) {
```
Comment (wburns) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HitsAwareCacheManagersTest.java]: Need to remove this

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HitsAwareCacheManagersTest.java [hunk 2]
```diff
@@ -200,23 +201,27 @@ protected void destroy() {
...
 
       public int getHits() {
-         return invocationCount;
+         return invocationCount.get();
       }
 
       public void reset() {
-         invocationCount = 0;
+         invocationCount.set(0);
+         log.infof("Hit count cleared");
```
Comment (wburns) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HitsAwareCacheManagersTest.java]: Remove as well

File: server/hotrod/src/test/java/org/infinispan/server/hotrod/test/HotRodTestingUtils.java [hunk 3]
```diff
@@ -138,7 +159,7 @@ public static HotRodServer startHotRodServer(EmbeddedCacheManager manager, Strin
       return startHotRodServer(manager, host, port, delay, false, builder);
    }
 
-   public static HotRodServer startHotRodServer(EmbeddedCacheManager manager, String host, int port, long delay, Boolean perf, HotRodServerConfigurationBuilder builder) {
+   public static HotRodServer startHotRodServer(EmbeddedCacheManager manager, String host, int port, long delay, boolean perf, HotRodServerConfigurationBuilder builder) {
```
Comment (wburns) [inline server/hotrod/src/test/java/org/infinispan/server/hotrod/test/HotRodTestingUtils.java]: Thanks, darn scala conversion I missed

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HitsAwareCacheManagersTest.java (hunk 1 above)
Comment (danberindei) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HitsAwareCacheManagersTest.java]: @wburns Sebastian already integrated it, but I'm opening a follow-up PR to remove this and convert the interceptor to `BaseAsyncInterceptor`.

**Comment by wburns** [comment]
Sounds great!

**Pull request merged**